src/us/kbase/narrativemethodstore/db/NarrativeMethodData.java

```diff
@@ -532,8 +532,8 @@ else if(shouldAdd) {
 			}
 			DropdownOptions ddOpt = null;
 			if (paramNode.has("dropdown_options")) {
-				JsonNode optNode = get(paramPath, paramNode, "dropdown_options");
-				optNode = get(paramPath + "/dropdown_options", optNode, "options");
+				JsonNode ddOptNode = get(paramPath, paramNode, "dropdown_options");
+				JsonNode optNode = get(paramPath + "/dropdown_options", ddOptNode, "options");
```

Comment:
i don't see any test changes - does this change the behavior at all?

Reply:
I'm going to add a couple of tests -- it took a while to track down where the test data for the server tests (`FullServerTest.java`) was coming from (of course it wasn't documented - docs are for noobs!), but now I've got it I can add in a few tests.

Reply:
OK, added a set of tests to `FullServerTest.java` to ensure that this edit was correct. I also ran the tests on the code on the pre-edit version to verify that they would fail. [They did](https://travis-ci.org/github/kbase/narrative_method_store/builds/730331035).

Reply:
There's tests in there now. @MrCreosote - any other requests?

Reply:
There's a `NarrativeMethodDataTest` class that handles unit tests for the chunk of code changed in this PR, which is a lot easier to deal with because you don't need to add test specs in another repo. You could consider moving your tests or adding tests there, but I'm good with what exists now if not.

Also the NMDT class test setup is quite a bit more sane than the full server tests.

Reply:
I've already added tests to that class (in the previous PR); this set adds to those existing tests.

Reply:
So did the tests not cover the behavior changed here? I would've naively expected unit test changes based on the behavior change. If not, should they?

Reply:
Oh happy day! Those tests were not being executed. 😖

Reply:
I've added the missing test file, plus a couple of others in the same directory, to the `build.xml`, so now they will actually be run!

Reply:
That test file is 4 years old...

Reply:
Ok, so last question I hope. I'm guessing the narrative method data tests failed prior to this fix? I don't see any behavior changes in the test file, hence the guess.

Reply:
Or did the tests pass erroneously because they loaded test case 8 vs 12

Reply:
There were three test files that weren't being run at all, including the `NarrativeMethodDataTest.java` file, which is why altering the code in `NarrativeMethodData.java` seemingly had no effect on what the tests did. When I enabled the tests in `NarrativeMethodDataTest.java`, it turned out that test case 8 has an (unrelated) error, so I created a new set of test files for the dropdown tests so as not to mess up any existing tests. I did [an additional run of testing with deliberate test failures](https://travis-ci.org/github/kbase/narrative_method_store/pull_requests) in `NarrativeMethodDataTest.java` to ensure that there wasn't anything else strange going on.